mayastor/src/lvm/lvm.rs

```diff
@@ -0,0 +1,285 @@
+                state: PoolState::PoolOnline.into(),
+            });
+        }
+        None => None
+    }
+}
+
+pub async fn import_or_create_lvm_vg(req: CreatePoolRequest) -> Result<Pool, Error> {
+    let name = req.name.as_str();
+    let disks = req.disks.as_slice();
+    let pool = match import_lvm_vg(CreatePoolRequest{
+            name: name.to_string(),
+            disks: disks.to_owned(),
+            pooltype: PoolType::Lvm.into(),
+        }).await {
+        Ok(p) => p,
+        Err(e) => match e {
+            Error::NotFound{..} => create_lvm_vg(CreatePoolRequest{
+                name: name.to_string(),
+                disks: disks.to_owned(),
+                pooltype: PoolType::Lvm.into(),
+            }).await?,
+            _ => return Err(e)
+        }
+    };
+    info!("The pool '{}' has been created on {:?}", name, disks);
+    Ok(pool)
+}
+
+pub async fn import_lvm_vg(req: CreatePoolRequest) -> Result<Pool, Error> {
+    let pools = list_lvm_vg().await?;
+    match pools.iter().find(|p| p.name == req.name) {
+        Some(p) => {
+            if p.disks == req.disks {
+                return Ok(Pool{
+                    name: p.name.as_str().to_owned(),
+                    disks: p.disks.as_slice().to_owned(),
+                    capacity: p.capacity,
+                    used: p.used,
+                    pooltype: PoolType::Lvm.into(),
+                    state: PoolState::PoolOnline.into(),
+                });
+            }
+            Err(Error::ImportErr {err: format!("lvm vg {} disks not match", req.name)})
+        }
+        None => Err(Error::NotFound {err: format!("lvm volume {} not found", req.name)}),
+    }
+}
+
+pub async fn create_lvm_vg(req: CreatePoolRequest) -> Result<Pool, Error> {
+    let output = Command::new(PVCREATE_COMMAND)
+        .args(&req.disks)
+        .output()
+        .await
+        .expect("failed to execute pv_create");
+    if !output.status.success() {
+        let msg = match String::from_utf8(output.stderr){
+            Ok(s) => s,
+            Err(_) => "failed to parse stderr for pv_create".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg})
+    }
+
+    let output = Command::new(VGCREATE_COMMAND)
+        .arg(req.name.as_str())
+        .args(&req.disks)
+        .arg("--addtag=mayastor")
+        .output()
+        .await
+        .expect("failed to execute vg_create");
+    if !output.status.success() {
+        let msg = match String::from_utf8(output.stderr){
+            Ok(s) => s,
+            Err(_) => "failed to execute vg_create".to_string(),
+        };
+        return Err(Error::FailedExec{err: msg.to_string()})
+    }
+
+    get_lvm_vg(req.name).await
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct VGListReport {
+    report: Vec<VGList>
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VGList {
+    vg: Vec<VGName>
+}
+#[derive(Debug, Serialize, Deserialize)]
+struct VGName {
+    vg_name: String
+}
```

Comment:
Since these types don't really have an impl (i-e they don't abstract the underlying type) and are private even, they're good candidates for type alias instead:

```suggestion
type VolGroupNameList = Vec<VolGroupName>;
type VolGroupName = String;
```

(note: i'm assuming descriptive names as per my previous suggestions here).

Reply:
I am not sure but we need the same name as the son fields for `serde.rs` to work. The json returned by this command would look like 

```
$ sudo vgs --options=vg_name --reportformat=json @mayastor
  {
      "report": [
          {
              "vg": [
                  {"vg_name":"pool"}
              ]
          }
      ]
  }
``` 
The struct names can be improved as suggested but I guess the fields should remain the same.

Reply:
Ah, sorry I missed on that detail. :)